test/dynamo/test_functions.py

```diff
@@ -1541,6 +1541,73 @@ def add_kernel(
         # Make sure it is NOT modified
         self.assertEqual(output, torch.zeros_like(t1))
 
+    @requires_cuda()
+    @requires_triton()
+    def test_triton_kernel_functionalize(self):
```

Comment:
I think these tests are fine for this PR, but when we add inductor support it would be great to have some more comprehensive tests (just testing that the outputs match eager mode):

(1) triton kernel that mutates an input to the graph
(2) triton kernel that mutate a view of an input to the graph
(3) graph that returns a view/alias of one of the mutated arguments to a triton kernel

For the first case (simple input mutation), there are ~3 individual cases worth testing. This is because these cases impact whether or not AOTAutograd sends the `copy_()` (input mutation) directly to inductor in the graph, so they affect whether or not inductor will be able to optimize away the clone.

(a) the graph is an inference graph (test is under `no_grad()`, or no inputs to the graph require gradient)
(b) graph is a training graph (at least one input requires grad), **but** the arguments to the triton kernel to not require grad
(c) graph is a training graph, and the argument(s) to the triton kernel also require grad.

Reply:
Will add as follow up. Although I need some pointers for how to implement some of these test cases. 

Right now all my tests are (1) I believe and none of my tensors require grad (i did not do require_grad=True).
* How do make inputs to triton kernel require grad?
* Do I need to test things differently when tensors require grad or do I still check outputs match?

Reply:
> How do make inputs to triton kernel require grad?

You can just set your tensor inputs to the compiled code to require grad:
```
@torch.compile
def f(x):
    return kernel[grid](x, ...)

x = torch.randn(4, requires_grad=True)
out = f(x)
```

> Do I need to test things differently when tensors require grad or do I still check outputs match?

Testing just the forward is fine - but you're right, a more comprehensive test would involve also running autograd and checking that the gradients match between eager and compile. Example:
```
@torch.compile
def f_triton_input_mutation_of_view(x, out):
    out_view = out[:, 1]
    kernel[grid](x=x, out=out_view)
    return torch.mul(out_view, 2)

x = torch.randn(4)
out = torch.randn(4, requires_grad=True)
# run autograd
out.sum().backward()

# Run the same with ref inputs on eager, check that the grads match
self.assertEqual(out.grad, out_ref.grad)
```